### PR TITLE
Add wicketd-client

### PR DESCRIPTION
**Pull request by andrewjstone**
Add a progenitor generated-client using the wicketd openapi spec.

The structure of the output for the `/inventory` endpoint was changed since tuples are unsupported. While doing this, I prepared for further changes that will include a `Vec<SpComponentInfo>` per SP. That code is currently awaiting some progenitor additions to allow deriving types like `Eq, Ord`, etc... on arbitrary types so that wicket can reuse MGS types directly.

While waiting for those changes, I'll begin integrating wicket and wicketd.

### CODE REVIEW

File: openapi/wicketd.json [hunk 1]
```diff
@@ -0,0 +1,349 @@
+{
```
Comment (jgallagher) [inline openapi/wicketd.json:1]: Could we add an integration test that confirms this file stays in sync? The MGS one is at https://github.com/oxidecomputer/omicron/blob/adefcef27cc074f7c55cc4c0f12adee49eac3e5b/sled-agent/tests/integration_tests/commands.rs#L58-L79

**Review by jgallagher — APPROVED** [review]

File: openapi/wicketd.json (hunk 1 above)
Comment (andrewjstone) [reply openapi/wicketd.json:1]: Done in [f3aa8da](https://github.com/oxidecomputer/omicron/pull/1930/commits/f3aa8da39612614079ee2bf6d03ec179d0489e39)

File: wicketd/tests/integration_tests/commands.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+use std::path::PathBuf;
+
+use expectorate::assert_contents;
+use omicron_test_utils::dev::test_cmds::{
+    assert_exit_code, path_to_executable, run_command, EXIT_SUCCESS,
+};
+use openapiv3::OpenAPI;
+use subprocess::Exec;
+
+// name of mgs executable
```
Comment (jgallagher) [inline wicketd/tests/integration_tests/commands.rs:18]:
```suggestion
// name of wicketd executable
```

(or remove the comment altogether; doesn't seem super valuable)

File: wicketd/tests/integration_tests/commands.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+const CMD_WICKETD: &str = env!("CARGO_BIN_EXE_wicketd");
+
+fn path_to_mgs() -> PathBuf {
```
Comment (jgallagher) [inline wicketd/tests/integration_tests/commands.rs:21]:
```suggestion
fn path_to_wicketd() -> PathBuf {
```

File: wicketd/tests/integration_tests/commands.rs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    path_to_executable(CMD_WICKETD)
+}
+
+#[test]
+fn test_wicketd_openapi() {
+    let exec = Exec::cmd(path_to_mgs()).arg("openapi");
```
Comment (jgallagher) [inline wicketd/tests/integration_tests/commands.rs:27]:
```suggestion
    let exec = Exec::cmd(path_to_wicketd()).arg("openapi");
```

**Comment by andrewjstone** [comment]
Thanks for catching all my copy-paste errors @jgallagher

**Pull request merged**